### PR TITLE
chore: add model.go ut

### PR DESCRIPTION
**Pull request by yyzxw**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/model.go [hunk 1]
```diff
@@ -83,7 +87,11 @@ func (refs AllowedReferences) BackendAllowed(
 ) bool {
 	from := Reference{Kind: k, Namespace: k8s.Namespace(routeNamespace)}
 	to := Reference{Kind: gvk.Service, Namespace: backendNamespace}
-	allow := refs[from][to]
+	mFromMap, ok := refs[from]
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/model.go:90]: why this change? it doesn't seem to be fixing any bugs or making the code simpler

Comment (yyzxw) [reply pilot/pkg/config/kube/gateway/model.go:90]: It may cause a NPE

Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/model.go:90]: I don't think that is true, read of nil map is ok.

Comment (yyzxw) [reply pilot/pkg/config/kube/gateway/model.go]: yes, i reverted it, thanks for your review!

**Comment by hzxuzhonghu** [comment]
isn't there already a test `TestReferencePolicy` tested this function
